src/extension/intellisenseHelper.ts

```diff
@@ -53,21 +69,53 @@ export class IntellisenseHelper {
     }
 
     /**
-     * Helper method that installs Typescript into a global location.
+     * Helper method that verifies the correct version of TypeScript is installed.
+     * If using a newer version of VSCode TypeScript is installed by default and no
+     * action is needed. If using an older version, verify that the correct TS version is
+     * installed, if not install it.
      */
-    public static installTypescriptNext(isRestartRequired: boolean): Q.Promise<boolean> {
-        let typeScriptNextDest: string = path.resolve(IntellisenseHelper.getUserHomePath(), ".vscode");
-        let typeScriptNextLibPath: string = path.join(typeScriptNextDest, "node_modules", "typescript", "lib");
-        let fileSystem: FileSystem = new FileSystem();
+    public static verifyInstallTypeScript(isRestartRequired: boolean): Q.Promise<boolean> {
 
-        return fileSystem.exists(typeScriptNextLibPath)
-            .then(function(exists: boolean) {
-                if (!exists) {
-                    return Q.nfcall(child_process.exec, `npm install --prefix ${typeScriptNextDest} typescript@next`)
-                        .then(() => { return true; });
+        if (IntellisenseHelper.isSalsaSupported()) {
+            // this is the correct version of vscode, which includes TypeScript (Salsa) support, nothing to do here
+            return Q.resolve<boolean>(isRestartRequired);
+        }
+
+        return IntellisenseHelper.getInstalledTypeScriptVersion()
+            .then(function(installProps: InstallProps) {
+
+                if (installProps.installed === false) {
+
+                    if(semver.neq(IntellisenseHelper.s_typeScriptVersion, installProps.version)) {
```

Comment:
My thought was that since this isn't an official way to do this, we should stick to one version that we test (the release version). Since VSCode will ship native support for Salsa it will be on them to define what versions they want to test and support.